File: 0704-binary-search/0704-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int str =0;
        int end = nums.length-1;
        int ans = -1;
        while(str <= end){
            int mid = str+(end-str)/2;
            if(target == nums[mid]){
                return mid;
            }
            else if(target > nums[mid]){
                str = mid+1;
            }
            else{
                end = mid-1;
            }
        }
        return ans;
    }
}

File: 0704-binary-search/0704-binary-search.cpp
class Solution {
public:
    int binSearch(vector<int>& arr,int target,int str,int end){
        if(str>end){
            return -1;
        }
       int mid = str+(end-str)/2;
       if(arr[mid]==target){
       return mid;
       }
       if(arr[mid]>target){
            return binSearch(arr,target,str,mid-1);
       }
    //    else if(arr[mid]<target){
        return binSearch(arr,target,mid+1,end);
    }
    int search(vector<int>& nums, int target) {
        return binSearch(nums,target,0,nums.size()-1);
    }
};